cmd/changelog-extractor/main.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright 2019 Splunk Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+	if len(os.Args) != 2 {
+		log.Fatalln("Must pass the Changelog file name as the first argument")
+	}
+	filename := os.Args[1]
+	if err := printReleaseNotes(filename); err != nil {
+		log.Fatalln(err)
+	}
+
+}
+
+func printReleaseNotes(filename string) error {
```

Comment:
should this also take an `expected version` string and fail fast if that version doesn't exist?

Reply:
consider the case where a maintainer tags a release but forgets to update the changelog. It shouldn't use the older tag by default.

Reply:
I thought about this case but left it out for now. I'll add that enhancement. There's also a minor issue handling multi-lines for bullets: https://github.com/harsimranmaan/qbec/releases/tag/v0.8.2. 
I'll address them when I revisit this in the next few days. TBH, Changlog parsing and publishing could be it's own project 😉